### PR TITLE
[vcpkg] Implement a warning when running in a mismatched Visual Studio prompt

### PR DESCRIPTION
**Pull request by BillyONeal**
Example output:

image

### CODE REVIEW

File: toolsrc/src/vcpkg-test/system.cpp [hunk 1]
```diff
@@ -122,6 +122,11 @@ TEST_CASE ("[from_cpu_architecture]", "system")
 TEST_CASE ("[guess_visual_studio_prompt]", "system")
```
Comment (strega-nil) [inline toolsrc/src/vcpkg-test/system.cpp]: This should probably be:

```cpp
TEST_CASE("guess visual studio prompt", "[system]")
```
```

**Comment by strega-nil** [comment]
Other than that, looks good to me :)

**Comment by BillyONeal** [comment]
@strega-nil What do you think about the "only emit when installing new packages" decision? i.e. I could see an argument to emit the warning when everything is already installed too...

**Comment by strega-nil** [comment]
@BillyONeal I'm personally on the fence here; I'm probably more on the "always print the warning" side, but I can see the benefits to both.

**Pull request merged**